From dd640e7e0cca676886c55803ef2bcec3e8068d50 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Wed, 8 Aug 2018 13:34:33 +0800
Subject: [PATCH 4404/5242] MLK-19155: ASoC: fsl_rpmsg: fix the volume is low
 for S24_LE

commit  88b8db3e9f42fcedbac7b69300ab94cf56cae501 from
https://source.codeaurora.org/external/imx/linux-imx.git

The format send to M4 through rpmsg is wrong, that make the
driver treat the data as S32_LE, it looks like data is right
shift 8 bit.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
(cherry picked from commit 508550b70e80339d3d49594ffc23946dd80b0c82)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_rpmsg_i2s.c |    3 +--
 sound/soc/fsl/imx-pcm-rpmsg.c |    2 +-
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/sound/soc/fsl/fsl_rpmsg_i2s.c b/sound/soc/fsl/fsl_rpmsg_i2s.c
index 4d004a3..202906c 100644
--- a/sound/soc/fsl/fsl_rpmsg_i2s.c
+++ b/sound/soc/fsl/fsl_rpmsg_i2s.c
@@ -192,8 +192,7 @@ static int fsl_rpmsg_i2s_probe(struct platform_device *pdev)
 				SNDRV_PCM_RATE_192000;
 		rpmsg_i2s->formats = SNDRV_PCM_FMTBIT_S16_LE |
 					SNDRV_PCM_FMTBIT_S24_LE |
-					SNDRV_PCM_FMTBIT_S32_LE |
-					SNDRV_PCM_FMTBIT_S24_3LE;
+					SNDRV_PCM_FMTBIT_S32_LE;
 		fsl_rpmsg_i2s_dai.playback.rates = rpmsg_i2s->rates;
 		fsl_rpmsg_i2s_dai.playback.formats = rpmsg_i2s->formats;
 		fsl_rpmsg_i2s_dai.capture.rates = rpmsg_i2s->rates;
diff --git a/sound/soc/fsl/imx-pcm-rpmsg.c b/sound/soc/fsl/imx-pcm-rpmsg.c
index a17f6cf..5a46e1f 100644
--- a/sound/soc/fsl/imx-pcm-rpmsg.c
+++ b/sound/soc/fsl/imx-pcm-rpmsg.c
@@ -63,7 +63,7 @@ static int imx_rpmsg_pcm_hw_params(struct snd_pcm_substream *substream,
 
 	if (params_format(params) == SNDRV_PCM_FORMAT_S16_LE)
 		rpmsg->send_msg.param.format   = RPMSG_S16_LE;
-	else if (params_format(params) == SNDRV_PCM_FORMAT_S24_3LE)
+	else if (params_format(params) == SNDRV_PCM_FORMAT_S24_LE)
 		rpmsg->send_msg.param.format   = RPMSG_S24_LE;
 	else
 		rpmsg->send_msg.param.format   = RPMSG_S32_LE;
-- 
1.7.9.5

